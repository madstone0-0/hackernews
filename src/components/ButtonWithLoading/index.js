/* eslint-disable no-unused-vars */
import Loading from "../Loading";
import Button from "../Button";
import React from "react";
import PropTypes from "prop-types";

const withLoading = Component => ({ isLoading, ...rest }) =>
    isLoading ? <Loading/> : <Component {...rest} />;

const ButtonWithLoading = withLoading(Button);

export default ButtonWithLoading;

ButtonWithLoading.propTypes = {
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
};
