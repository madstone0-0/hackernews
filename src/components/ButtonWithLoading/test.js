/* eslint-disable no-unused-vars,no-undef,no-console */
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import React from "react";
import ButtonWithLoading from "./index";

describe("ButtonWithLoading", () => {
    it("renders", () => {
        const div = document.createElement("div");
        ReactDOM.render(<ButtonWithLoading onClick={() => {
            console.log("Clicked");
        }}>Loading Button</ButtonWithLoading>, div);

    });
});

test("snapshots", () => {
    const component = renderer.create(
        <ButtonWithLoading onClick={() => {
            console.log("Clicked");
        }}> Button </ButtonWithLoading>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});