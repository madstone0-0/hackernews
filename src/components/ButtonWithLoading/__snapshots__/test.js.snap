// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
   Button 
</button>
`;
